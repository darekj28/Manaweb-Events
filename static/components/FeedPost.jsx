var React = require('react');
var Link = require('react-router').Link;
// var $ = require('jquery');
import Avatar from "./Avatar.jsx";
import FeedPostHeader from "./FeedPostHeader.jsx";
import FeedPostBody from "./FeedPostBody.jsx";

export default class FeedPost extends React.Component {	
	constructor(props) {
		super(props);
		this.state = {comment_id : this.props.post.comment_id};
		this.handlePostEdit = this.handlePostEdit.bind(this);
		this.handlePostDelete = this.handlePostDelete.bind(this);
		this.handlePostReport = this.handlePostReport.bind(this);
		this.scrollToDropdown = this.scrollToDropdown.bind(this);
	}
	handlePostEdit() {
		this.props.refreshPostDisplayedInModal(this.props.post);
		$('#EditPostModal').modal('show');
	}
	handlePostDelete() {
		this.props.refreshPostDisplayedInModal(this.props.post);
		$('#DeletePostModal').modal('show'); 
	}
	handlePostReport() {
		this.props.refreshPostDisplayedInModal(this.props.post);
		$('#ReportPostModal').modal('show');
	}
	componentWillReceiveProps(nextProps) {
		this.setState({ comment_id : nextProps.post.comment_id });
	}
	scrollToDropdown() {
		var post = document.getElementById('post_' + this.props.post.comment_id);
		var offset = post.offsetTop - 300;
		$('#Feed').animate({scrollTop : offset}, 500);
	}
	render() {
		var post = this.props.post;
		return (
			<li className="Post" id = {"post_" + post.comment_id}>
				<Avatar source={post.avatar}/>
				<div className="PostSpace">
					<div className="row"><FeedPostHeader post={this.props.post} 
						handleFilterUser = {this.props.handleFilterUser} 
						isAdmin={this.props.isAdmin} isOP={this.props.isOP}/></div>
					<div className="row"><FeedPostBody content={post.postContent}/></div>
					<div className="PostFooter row">
						<div className="CommentContainer">
						<Link to={'/comment/' + this.state.comment_id}>
						<span className="glyphicon glyphicon-comment pull-left PostBottomIcon AppGlyphicon" 
								id={"viewComment_" + this.state.comment_id}>
								</span>
						<span className="numberOfComments pull-left AppGlyphicon"><h6>{post.numberOfComments}</h6></span>
						</Link>
						</div>
						{ (this.props.isAdmin || !this.props.isOP) && 
						<div className="dropdown pull-right" id={"dropdown_" + this.props.post.comment_id} onClick={this.scrollToDropdown}>
							<a href="#" className="dropdown-toggle" data-toggle="dropdown">
				                <span className="glyphicon glyphicon-option-horizontal 
				                				pull-right PostBottomIcon AppGlyphicon"></span>
				            </a>
				            <ul className="PostDropdown pull-right dropdown-menu">
				              	{(this.props.isAdmin) && <li><a id="hpe" href="#" onClick={this.handlePostEdit}>Edit post</a></li> }
			              		{(this.props.isAdmin) && <li><a id="hpd" href="#" onClick={this.handlePostDelete}>Delete post</a></li> }
			              		{(!this.props.isOP || this.props.isAdmin) && <li><a id="hpr" href="#" onClick={this.handlePostReport}>Report post</a></li> }
				            </ul>
				        </div>}	
					</div>
				</div>
			</li>
		);
	}
}