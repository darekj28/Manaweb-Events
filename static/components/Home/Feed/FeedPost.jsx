var React = require('react');
var Link = require('react-router').Link;
import Avatar from "./Avatar.jsx";
import FeedPostHeader from "./FeedPostHeader.jsx";
import FeedPostBody from "./FeedPostBody.jsx";

export default class FeedPost extends React.Component {	
	constructor(props) {
		super(props);
		this.state = {comment_id : this.props.post.comment_id};
	}
	handlePostEdit() {
		this.props.refreshPostDisplayedInModal(this.props.post);
		$('#EditPostModal').modal('show');
	}
	handlePostDelete() {
		var callback = function() {
			swal({title : "Success!", 
						text: "The post has been deleted.", 
						type: "success",
						confirmButtonColor: "#80CCEE",
		  				confirmButtonText: "OK"
					});
		}
		swal({
	      	title: "Are you sure you want to delete this post?", 
		    showCancelButton: true,
	      	closeOnConfirm: false,
	      	confirmButtonText: "Yes, I'm sure!",
	      	confirmButtonColor: "#80CCEE"}, 
	    	function() {
	    	var obj = {unique_id : this.props.post.unique_id,
					jwt: localStorage.jwt};
			$.ajax({
				type : 'POST',
				url  : '/deletePost',
				data : JSON.stringify(obj, null, '\t'),
				contentType: 'application/json;charset=UTF-8',
				success: function(data) {
					this.props.handlePostDelete(this.props.post, callback);
				}.bind(this)
			});
	    }.bind(this));
	}
	handlePostReport() {
		this.props.refreshPostDisplayedInModal(this.props.post);
		$('#ReportPostModal').modal('show');
	}
	componentWillReceiveProps(nextProps) {
		this.setState({ comment_id : nextProps.post.comment_id });
	}
	scrollToDropdown() {
		var post = document.getElementById('dropdown_' + this.props.post.comment_id);
		var offset = post.offsetTop - 300;
		$('html, body').animate({scrollTop : offset}, 500);
	}
	render() {
		var post = this.props.post;
		var isOP = this.props.isOP;
		var isAdmin = this.props.isAdmin;
		var reply = post.numberOfComments == 0 || post.numberOfComments > 1 ? "replies" : "reply";
		return (
			<li className="Post" id = {"post_" + post.comment_id}>
				<Avatar source={post.avatar}/>
				<div className="PostSpace">
					<div className="row"><FeedPostHeader post={this.props.post} 
						handleFilterUser = {this.props.handleFilterUser} 
						isAdmin={isAdmin} isOP={isOP}/></div>
					<div className="row"><FeedPostBody content={post.postContent}/></div>
					<div className="PostFooter row">
						<div className="CommentContainer">
							<Link to={'/comment/' + this.state.comment_id}>
							<span className="glyphicon glyphicon-comment pull-left PostBottomIcon AppGlyphicon" 
									id={"viewComment_" + this.state.comment_id}>
									</span>
							{post.numberOfComments > 0 && <span className="numberOfComments pull-left AppGlyphicon"><h6>{post.numberOfComments} {reply}</h6></span>}
							{post.numberOfComments == 0 && <span className="noComments pull-left"><h6>Be the first to comment!</h6></span>}
							</Link>
						</div>
						{ ((isAdmin || !isOP) && post.userID != "$DELETED_USER") && 
						<div className="dropdown pull-right" id={"dropdown_" + post.comment_id} onClick={this.scrollToDropdown.bind(this)}>
							<a href="#" className="dropdown-toggle" data-toggle="dropdown">
				                <span className="glyphicon glyphicon-option-horizontal 
				                				pull-right PostBottomIcon AppGlyphicon"></span>
				            </a>
				            <ul className="PostDropdown pull-right dropdown-menu">
				              	{(isAdmin) && <li><a id="hpe" onClick={this.handlePostEdit.bind(this)}>Edit post</a></li> }
			              		{(!isOP || isAdmin) && <li><a id="hpr" onClick={this.handlePostReport.bind(this)}>Report post</a></li> }
			              		{(isAdmin) && <li><a id="hpd" onClick={this.handlePostDelete.bind(this)}>Delete post</a></li> }
				            </ul>
				        </div>}	
					</div>
				</div>
			</li>
		);
	}
}