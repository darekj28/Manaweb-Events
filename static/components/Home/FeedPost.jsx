var React = require('react');
var Link = require('react-router').Link;
import Avatar from "./Avatar.jsx";
import FeedPostHeader from "./FeedPostHeader.jsx";
import FeedPostBody from "./FeedPostBody.jsx";

export default class FeedPost extends React.Component {	
	constructor(props) {
		super(props);
		this.state = {comment_id : this.props.post.comment_id};
	}
	handlePostEdit() {
		this.props.refreshPostDisplayedInModal(this.props.post);
		$('#EditPostModal').modal('show');
	}
	handlePostDelete() {
		this.props.refreshPostDisplayedInModal(this.props.post);
		$('#DeletePostModal').modal('show'); 
	}
	handlePostReport() {
		this.props.refreshPostDisplayedInModal(this.props.post);
		$('#ReportPostModal').modal('show');
	}
	componentWillReceiveProps(nextProps) {
		this.setState({ comment_id : nextProps.post.comment_id });
	}
	scrollToDropdown() {
		var post = document.getElementById('dropdown_' + this.props.post.comment_id);
		var offset = post.offsetTop - 300;
		$('html, body').animate({scrollTop : offset}, 500);
	}
	render() {
		var post = this.props.post;
		var isOP = this.props.isOP;
		var isAdmin = this.props.isAdmin;
		return (
			<li className="Post" id = {"post_" + post.comment_id}>
				<Avatar source={post.avatar}/>
				<div className="PostSpace">
					<div className="row"><FeedPostHeader post={this.props.post} 
						handleFilterUser = {this.props.handleFilterUser} 
						isAdmin={isAdmin} isOP={isOP}/></div>
					<div className="row"><FeedPostBody content={post.postContent}/></div>
					<div className="PostFooter row">
						<div className="CommentContainer">
							<Link to={'/comment/' + this.state.comment_id}>
							<span className="glyphicon glyphicon-comment pull-left PostBottomIcon AppGlyphicon" 
									id={"viewComment_" + this.state.comment_id}>
									</span>
							<span className="numberOfComments pull-left AppGlyphicon"><h6>{post.numberOfComments}</h6></span>
							</Link>
						</div>
						{ (isAdmin || !isOP) && 
						<div className="dropdown pull-right" id={"dropdown_" + post.comment_id} onClick={this.scrollToDropdown.bind(this)}>
							<a href="#" className="dropdown-toggle" data-toggle="dropdown">
				                <span className="glyphicon glyphicon-option-horizontal 
				                				pull-right PostBottomIcon AppGlyphicon"></span>
				            </a>
				            <ul className="PostDropdown pull-right dropdown-menu">
				              	{(isAdmin) && <li><a id="hpe" onClick={this.handlePostEdit.bind(this)}>Edit post</a></li> }
			              		{(isAdmin) && <li><a id="hpd" onClick={this.handlePostDelete.bind(this)}>Delete post</a></li> }
			              		{(!isOP || isAdmin) && <li><a id="hpr" onClick={this.handlePostReport.bind(this)}>Report post</a></li> }
				            </ul>
				        </div>}	
					</div>
				</div>
			</li>
		);
	}
}