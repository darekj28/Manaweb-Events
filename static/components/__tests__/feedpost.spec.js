import FeedPost from "../FeedPost.jsx";

describe("FeedPost", () => {
	it("renders correctly", () => {
		const post = {comment_id : "a", name : 'a', userID : 'a', time : 'a',
						isTrade : true, isPlay : true, isChill : true,
						postContent : "a", avatar : "a"};
		const fp = renderer.create(
			<FeedPost post={post} isOP={true} isAdmin={true}
					refreshFeed={jest.fn()} 
					refreshPostDisplayedInModal={jest.fn()}
					handleFilterUser={jest.fn()} />
			);
		expect(fp).toMatchSnapshot();
	});
	it("should call handlePostEdit when edit post is clicked", () => {
		expect(true).toBe(true);
	});
	it("should call handlePostDelete when delete post is clicked", () => {
		expect(true).toBe(true);
	});
	it("should call handlePostReport when report post is clicked", () => {
		expect(true).toBe(true);
	});
});