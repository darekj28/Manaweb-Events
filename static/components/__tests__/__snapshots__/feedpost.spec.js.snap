exports[`FeedPost renders correctly 1`] = `
<li
  className="Post">
  <a
    className="pull-left AvatarSpace"
    href="#">
    <img
      className="Avatar img-rounded"
      src="a" />
  </a>
  <div
    className="PostSpace">
    <div
      className="row">
      <div
        className="FeedPostHeader">
        <div
          className="tpc pull-right">
          <kbd>
             Trade  Play  Chill 
          </kbd>
        </div>
        <div
          className="headerpart name"
          onClick={[Function]}>
          <b>
            a
          </b>
        </div>
        <div
          className="headerpart username text-muted"
          onClick={[Function]}>
          @
          a
        </div>
        <div
          className="headerpart time text-muted">
          â€¢ 
          a
        </div>
      </div>
    </div>
    <div
      className="row">
      <div
        className="FeedPostBody">
        a
      </div>
    </div>
    <div
      className="PostFooter row">
      <a
        onClick={[Function]}
        style={Object {}}>
        <span
          className="glyphicon glyphicon-comment pull-left PostBottomIcon AppGlyphicon"
          id="viewComment_a" />
      </a>
      <div
        className="dropdown">
        <a
          className="dropdown-toggle"
          data-toggle="dropdown"
          href="#">
          <span
            className="glyphicon glyphicon-option-horizontal  pull-left PostBottomIcon AppGlyphicon" />
        </a>
        <ul
          className="PostDropdown pull-left dropdown-menu">
          <li>
            <a
              href="#"
              id="hpe"
              onClick={[Function]}>
              Edit post
            </a>
          </li>
          <li>
            <a
              href="#"
              id="hpd"
              onClick={[Function]}>
              Delete post
            </a>
          </li>
        </ul>
      </div>
    </div>
  </div>
</li>
`;
